### DISCUSSION TITLE
Progress on VMU support?

### DISCUSSION THREAD

**Discussion by polyv0x** [darthcloud/BlueRetro#393] [General]
First of all I want to thank you on your fantastisc work so far on this project! It's quite amazing!
I have been testing with my ESP32 on a Dreamcast, and I ran into a problem where games that expect the VMU to be in port 1 are not able to save at all when using the BlueRetro controller. (And of course they also do not support controlling the game from any other port than port 1.) 

This is of course not a problem of BlueRetro, but the game itself. But it's common enough among games that it's at least troublesome. (Example in question; Shenmue 1)

So my question is, have you made any progress on the progress on VMU support? :)

**Comment by darthcloud** [comment]
No progress, not started.

I'm busy with official internal install currently.

And after that I got some fundamental investigation/work to do on the Bluetooth stack to fix various issues.

So VMU support is still far away, as I fully expect it to be a time consuming rabbit hole.

**Reply by polyv0x** [reply]
Thanks for the reply!

There is a similar project on Github (albeit not with bluetooth, but for the VMU part of things that shouldn't matter I suppose) for the pi pico called Maple Pad, which has a working switchable bank system as you are/were looking to implement. 
[https://github.com/mackieks/MaplePad](url)

I'm not sure how similar the underlying architecture of the platforms is but perhaps it may be of some help in your future endeavors!

Thank you for all the hard work and congrats on becoming a father :) I'm sure it's complicated already complicated life even further haha.
